lib/saluki-env/src/autodiscovery/mod.rs

```diff
@@ -59,19 +62,65 @@ pub struct AdvancedADIdentifier {
     pub kube_endpoints: Option<KubeNamespacedName>,
 }
 
-/// Configuration data
+trait RawData {
+    fn get_value(&self) -> &HashMap<MetaString, serde_yaml::Value>;
+
+    fn get(&self, key: &str) -> Option<&serde_yaml::Value> {
+        self.get_value().get(key)
+    }
+
+    fn to_bytes(&self) -> Result<Vec<u8>, GenericError> {
```

Comment:
I might suggest calling this `to_yaml` just to be explicit, but that could happen in a follow-up PR.